Gem/Code/Source/Components/Multiplayer/PlayerJumpPadEffectComponent.cpp

```diff
@@ -0,0 +1,100 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project. For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#include <Multiplayer/Components/NetworkCharacterComponent.h>
+#include <Multiplayer/Components/NetworkTransformComponent.h>
+#include <Source/Components/Multiplayer/PlayerJumpPadEffectComponent.h>
+
+namespace MultiplayerSample
+{
+    void PlayerJumpPadEffectComponent::Reflect(AZ::ReflectContext* context)
+    {
+        AZ::SerializeContext* serializeContext = azrtti_cast<AZ::SerializeContext*>(context);
+        if (serializeContext)
+        {
+            serializeContext->Class<PlayerJumpPadEffectComponent, PlayerJumpPadEffectComponentBase>()
+                ->Version(1);
+        }
+        PlayerJumpPadEffectComponentBase::Reflect(context);
+    }
+
+    void PlayerJumpPadEffectComponent::OnInit()
+    {
+    }
+
+    void PlayerJumpPadEffectComponent::OnActivate([[maybe_unused]] Multiplayer::EntityIsMigrating entityIsMigrating)
+    {
+    }
+
+    void PlayerJumpPadEffectComponent::OnDeactivate([[maybe_unused]] Multiplayer::EntityIsMigrating entityIsMigrating)
+    {
+    }
+
+    void PlayerJumpPadEffectComponent::ApplyJumpPadEffect(const AZ::Vector3& jumpVelocity,
+        const AZ::TimeMs& effectDuration)
+    {
+        if (HasController())
+        {
+            static_cast<PlayerJumpPadEffectComponentController*>(GetController())->ApplyJumpPadEffect(jumpVelocity, effectDuration);
```

Comment:
Consider moving all this functionality to the controller itself.. have the controller detect the trigger intersection rather than the casting to access the controller, which is evil. evil, evil.